### DISCUSSION TITLE
Extracting Stratum Area from Historical STRS Design

### DISCUSSION THREAD

**Discussion by zoyafuso-NOAA** [afsc-gap-products/StationAllocationAIGOA#13] [General]
Just a quick way to see whether it makes a different whether you get your stratum area from the strata directly or from aggregating the station areas by strata. 

```
## Connect to VPN!
## Import Library
library(RODBC)

## Set Up Channel
channel <- RODBC::odbcConnect(dsn = "AFSC", 
                              uid = NULL, 
                              pwd = NULL, 
                              believeNRows = FALSE)

## Query area information from the strata directly
strata_q <- "select * from GOA.GOA_STRATA where SURVEY = 'GOA'"
goa_strata <- RODBC::sqlQuery(channel = channel, query = strata_q)
area_strata <- goa_strata[, c("SURVEY", "STRATUM", "AREA")]

## Query area information from the stations directly
grid_q <- "select * from GOA.GOAGRID_GIS" 
goa_grid <-  RODBC::sqlQuery(channel = channel, query = grid_q)
area_station <- aggregate(AREA_KM2 ~ STRATUM, 
                          data = goa_grid, 
                          subset = STRATUM != 0, 
                          FUN = sum)

## Print differences
data.frame( stratum = area_station$STRATUM,
            area_km2_from_stns = area_station$AREA_KM2, 
            area_km2_from_strata = area_strata$AREA,
            diff_percent = 100 * (area_station$AREA_KM2 - area_strata$AREA) / 
              area_strata$AREA )[c(1:3, 57:59), ]
```

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ZACK~1.OYA/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ZACK~1.OYA/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

  | stratum | area_km2_from_stns | area_km2_from_strata | diff_percent
-- | -- | -- | -- | --
1 | 10 | 8331.828 | 8332.718 | -1.07E-02
2 | 11 | 13680.837 | 13681.241 | -2.96E-03
3 | 12 | 6877.686 | 6875.823 | 2.71E-02
57 | 530 | 3493.865 | 3493.863 | 5.69E-05
58 | 540 | 1887.426 | 1887.412 | 7.44E-04
59 | 550 | 1206.349 | 1206.344 | 4.01E-04

</body>

</html>

So very slightly different when you run it, probably due to the way the areas were calculated, projection issues, etc.

**Comment by Ned-Laman-NOAA** [comment]
Summing the areas from GOA_STRATA or from stations in GOAGRID_GIS by REGULATORY_AREA_NAME or by INPFC_AREA give similar results to what you saw above indicating that we are using only one set of boundaries for the two management zones that we're representing.

**Comment by zoyafuso-NOAA** [comment]
Thanks for checking that. The implication is that the biomass time series for REGULATORY_AREA_NAME (i.e. for WESTERN/CENTRAL/EASTERN GOA) will _for the most part_ be consistent between the historical and new designs. Different story if requesting for district-level biomass because the boundaries are now slightly different.
